src/System.Net.HttpListener/tests/HttpRequestStreamTests.cs

```diff
@@ -32,8 +36,59 @@ public void Dispose()
             _helper.Dispose();
         }
 
+        // Try to read 'length' bytes from stream or fail after timeout.
+        private async Task<int> GetBytesAsync(Func<byte[], int, int, Task<int>> cb, byte[] array, int offset, int length)
```

Comment:
I don't think this needs to be a delegate.  Just pass in the stream and call ReadAsync on it in the method.

I'd also name this ReadLengthAsync, or something like that.